src/lib/src/core/viewer-service/viewer.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable, NgZone, OnInit } from '@angular/core';
+import { Manifest } from '../models/manifest';
+import { Options } from '../models/options';
+
+declare const OpenSeadragon: any;
+@Injectable()
+export class ViewerService implements OnInit {
+  private readonly ZOOMFACTOR = 0.02;
+  private viewer: any;
+
+  constructor(private zone: NgZone) {}
+
+  ngOnInit(): void {}
+
+  setUpViewer(manifest: Manifest) {
+    if (manifest.tileSource) {
+      this.zone.runOutsideAngular(() => {
+        this.viewer = new OpenSeadragon.Viewer(Object.assign({}, new Options(manifest.tileSource)));
+      });
+      this.addToWindow();
+      this.addEvents();
+    }
+  }
+
+  getViewer() {
+    return this.viewer;
+  }
+
+  addToWindow() {
+    window.openSeadragonViewer = this.viewer;
+  }
+
+  destroy() {
+    if (this.viewer != null && this.viewer.isOpen()) {
+      this.viewer.destroy();
+    }
+  }
+
+  addEvents(): void {
+    this.addPinchEvents();
+  }
+
+  addPinchEvents(): void {
```

Comment:
This could be done like this:

OpenSeadragon({
    ...
    visibilityRatio: 1.0,
    constrainDuringPan: true,
    ...
});

ref: https://openseadragon.github.io/examples/ui-zoom-and-pan/